crates/shared/src/sources/balancer_v2/pool_fetching.rs

```diff
@@ -136,19 +137,47 @@ pub struct BalancerPoolFetcher {
     fetcher: Arc<dyn InternalPoolFetching>,
 }
 
+arg_enum! {
+    /// An enum containing all supported Balancer factory types.
+    #[derive(Clone, Copy, Debug)]
+    pub enum BalancerFactoryKind {
+        Weighted,
+        Weighted2Token,
```

Comment:
Fancy rename - I didn't know we could use numbers in var names!

Reply:
They just can't start with numbers. So:
- `a0` is valid
- `0a` is invalid